### PR TITLE
tests: msgq_api: Improve scenario testing

### PR DESCRIPTION
**Pull request by punitvara**
Pend multiple thread to wait for writing on msgq.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

File: tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c [hunk 1]
```diff
@@ -88,6 +88,36 @@ static void msgq_thread(struct k_msgq *pmsgq)
...
+
+	k_msgq_put(pmsgq, (void *)&data[0], K_FOREVER);
+
+	/**TESTPOINT: thread-thread data passing via message queue*/
+	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
+				      thread_entry_overflow, pmsgq, NULL, NULL,
+				      K_PRIO_PREEMPT(0),
+				      K_USER | K_INHERIT_PERMS, 0);
+
+	k_msgq_put(pmsgq, (void *)&data[1], K_FOREVER);
```
Comment (nashif) [inline tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c]: check for return value

File: tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c [hunk 2]
```diff
@@ -88,6 +88,36 @@ static void msgq_thread(struct k_msgq *pmsgq)
...
+	k_msgq_get(p1, &rx_buf[0], K_FOREVER);
+	k_msgq_get(p1, &rx_buf[1], K_FOREVER);
+
+	k_sem_give(&end_sema);
+}
+
+static void msgq_thread_overflow(struct k_msgq *pmsgq)
+{
+
+	k_msgq_put(pmsgq, (void *)&data[0], K_FOREVER);
```
Comment (nashif) [inline tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c]: check for return value

File: tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c [hunk 3]
```diff
@@ -88,6 +88,36 @@ static void msgq_thread(struct k_msgq *pmsgq)
 	purge_msgq(pmsgq);
 }
 
+static void thread_entry_overflow(void *p1, void *p2, void *p3)
+{
+	u32_t rx_buf[MSGQ_LEN];
+
+	k_msgq_get(p1, &rx_buf[0], K_FOREVER);
+	k_msgq_get(p1, &rx_buf[1], K_FOREVER);
```
Comment (nashif) [inline tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c]: check for return value

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
You are not testing return values for many of the APIs you are using. This should be part of the test, otherwise you are just executing code without checking for correct behaviour.

File: tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c (hunk 3 above)
Comment (punitvara) [reply tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c]: okay. Thanks for review @nashif . Changes has been pushed

File: tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c [hunk 4]
```diff
@@ -88,6 +88,48 @@ static void msgq_thread(struct k_msgq *pmsgq)
...
+	k_sem_give(&end_sema);
+}
+
+static void msgq_thread_overflow(struct k_msgq *pmsgq)
+{
+	int ret;
+
+	ret = k_msgq_put(pmsgq, (void *)&data[0], K_FOREVER);
+
+	zassert_false(ret, NULL);
```
Comment (nashif) [inline tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c]: please check for the actual return value you are expecting, with zassert_equal, not with zassert_true/false

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c (hunk 4 above)
Comment (punitvara) [reply tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c]: Done

**Review by nashif — APPROVED** [review]

**Pull request merged**